packages/cli/lib/sandboxes.js

```diff
@@ -120,26 +130,114 @@ const getSyncTypesWithContactRecordsPrompt = async (
+      const devSandboxLimit = usage['DEVELOPER'].limit;
+      const plural = devSandboxLimit !== 1;
+      const hasDevelopmentSandboxes = getHasSandboxesByType(
+        accountConfig,
+        DEVELOPER_SANDBOX
+      );
+      if (hasDevelopmentSandboxes) {
+        throw new Error(
+          i18n(
+            `cli.lib.sandbox.create.failure.alreadyInConfig.developer.${
+              plural ? 'other' : 'one'
+            }`,
+            {
+              accountName: accountConfig.name || accountId,
+              limit: devSandboxLimit,
+            }
+          )
+        );
+      } else {
+        const baseUrl = getHubSpotWebsiteOrigin(env);
+        throw new Error(
+          i18n(
+            `cli.lib.sandbox.create.failure.limit.developer.${
+              plural ? 'other' : 'one'
+            }`,
+            {
+              accountName: accountConfig.name || accountId,
+              limit: devSandboxLimit,
+              link: `${baseUrl}/sandboxes-developer/${accountId}/development`,
+            }
+          )
+        );
+      }
+    }
+  }
+  if (sandboxType === STANDARD_SANDBOX) {
+    if (usage['STANDARD'].available === 0) {
+      const standardSandboxLimit = usage['STANDARD'].limit;
+      const plural = standardSandboxLimit !== 1;
+      const hasStandardSandboxes = getHasSandboxesByType(
+        accountConfig,
+        STANDARD_SANDBOX
+      );
+      if (hasStandardSandboxes) {
+        throw new Error(
+          i18n(
+            `cli.lib.sandbox.create.failure.alreadyInConfig.standard.${
+              plural ? 'other' : 'one'
+            }`,
+            {
+              accountName: accountConfig.name || accountId,
+              limit: standardSandboxLimit,
+            }
+          )
+        );
+      } else {
+        const baseUrl = getHubSpotWebsiteOrigin(env);
+        throw new Error(
+          i18n(
+            `cli.lib.sandbox.create.failure.limit.standard.${
+              plural ? 'other' : 'one'
+            }`,
+            {
+              accountName: accountConfig.name || accountId,
+              limit: standardSandboxLimit,
+              link: `${baseUrl}/sandboxes-developer/${accountId}/standard`,
+            }
+          )
+        );
+      }
+    }
+  }
+};
+
 /**
  * @param {String} env - Environment (QA/Prod)
  * @param {Object} result - Sandbox instance returned from API
  * @param {Boolean} force - Force flag to skip prompt
  * @returns {String} validName saved into config
  */
 const saveSandboxToConfig = async (env, result, force = false) => {
-  // const configData = { env, personalAccessKey: result.personalAccessKey };
-  // TODO: Temporary, remove once PAK is generated on BE
-  const configData = await personalAccessKeyPrompt({
-    env,
-    account: result.sandbox.sandboxHubId,
-  });
-  // End temporary section
+  let configData = { env, personalAccessKey: result.personalAccessKey };
+  if (!result.personalAccessKey || result.personalAccessKey === 'test') {
```

Comment:
did you mean to leave this 'test' check in?

Reply:
Good catch - It was kinda necessary based on what we were returning from BE, but I think we're gonna get that keygen PR out today. Commits incoming